File: svntest/zul/src/org/zkoss/zul/Menuseparator.java
/* Menuseparator.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Thu Sep 22 10:59:00     2005, Created by tomyeh
}}IS_NOTE

Copyright (C) 2005 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 2.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/
package org.zkoss.zul;

import org.zkoss.zul.impl.XulElement;

/**
 * Used to create a separator between menu items.
 *
 *<p>Default {@link #getMoldSclass}: z-menu-separator. (since 3.5.0)
 * 
 * @author tomyeh
 */
public class Menuseparator extends XulElement {

	//-- Component --//
	public String getMoldSclass() {
		return _moldSclass == null ? "z-menu-separator" : super.getMoldSclass();
	}
	/** Not childable. */
	public boolean isChildable() {
		return false;
	}
}

File: zul/src/main/resources/web/js/zul/menu/Menuseparator.ts
/* Menuseparator.ts

	Purpose:

	Description:

	History:
		Thu Jan 15 09:02:35     2009, Created by jumperchen

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * Used to create a separator between menu items.
 *
 * @defaultValue {@link getZclass}: z-menuseparator.
 */
@zk.WrapClass('zul.menu.Menuseparator')
export class Menuseparator extends zul.Widget {
	/**
	 * @returns whether parent is a {@link Menupopup}
	 */
	isPopup(): boolean {
		return !!this.parent && this.parent instanceof zul.menu.Menupopup;
	}

	/**
	 * @returns the {@link Menubar} that contains this menuseparator, or null if not available.
	 */
	getMenubar(): zul.menu.Menubar | undefined {
		for (var p = this.parent; p; p = p.parent)
			if (p instanceof zul.menu.Menubar)
				return p;
		return undefined;
	}

	/** @internal */
	override doMouseOver_(evt: zk.Event): void {
		if (zul.menu._nOpen)
			zWatch.fire('onFloatUp', this); //notify all
		super.doMouseOver_(evt);
	}
}
